examples/nvram/read.c

```diff
@@ -124,17 +140,59 @@ int TPM2_NVRAM_Read_Example(void* userCtx, int argc, char *argv[])
     nv.handle.auth.size = auth.size;
     XMEMCPY(nv.handle.auth.buffer, auth.buffer, auth.size);
 
-    readSize = keyBlob.pub.size;
-    printf("Trying to read %d bytes of public key part from NV\n", readSize);
-    rc = wolfTPM2_NVReadAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
-        (byte*)&keyBlob.pub.publicArea, &readSize, 0);
-    if (rc != 0) goto exit;
+    if (partialRead != PRIVATE_PART_ONLY) {
+        readSize = sizeof(keyBlob.pub.size);
+        printf("Trying to read %d bytes of public key size marker\n", readSize);
+        rc = wolfTPM2_NVReadAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
+            (byte*)&keyBlob.pub.size, &readSize, 0);
+        if (rc != 0) {
+            printf("Was a public key part written? (see nvram/store)\n");
+            goto exit;
+        }
+        printf("Successfully read public key part from NV\n\n");
+        offset += readSize;
+
+       /* Prepare temporary buffer */
+        packet.buf = pubAreaBuffer;
+        packet.pos = 0;
+        packet.size = sizeof(pubAreaBuffer);
+
+        readSize = sizeof(UINT16) + keyBlob.pub.size; /* account for TPM2B size marker */
+        printf("Trying to read %d bytes of public key part from NV\n", keyBlob.pub.size);
+        rc = wolfTPM2_NVReadAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
+            packet.buf, &readSize, offset);
+        if (rc != 0) goto exit;
+        printf("Successfully read public key part from NV\n\n");
+        offset += readSize;
 
-    readSize = keyBlob.priv.size;
-    printf("Trying to read %d bytes of private key part from NV\n", readSize);
-    rc = wolfTPM2_NVReadAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
-        (byte*)&keyBlob.priv.buffer, &readSize, keyBlob.pub.size);
-    if (rc != 0) goto exit;
+        /* Necessary for storing the publicArea with the correct byte aligness */
+        TPM2_Packet_ParsePublic(&packet, &keyBlob.pub);
```

Comment:
This is not a public API. If bulding against the shared libwolftpm you will get:

```
Undefined symbols for architecture x86_64:
  "_TPM2_Packet_AppendPublic", referenced from:
      _writeKeyBlob in tpm_test_keys.o
  "_TPM2_Packet_ParsePublic", referenced from:
      _readKeyBlob in tpm_test_keys.o
```

Please add new public API to convert byte stream to `TPM2B_PUBLIC`

Reply:
understood. I will do the same for AppendPublic and ParsePublic